File: taller-5-java-ronalcastro97/src/main/java/zooAnimales/Anfibio.java
package zooAnimales;
import gestion.Zona;
import java.util.ArrayList;
public class Anfibio extends Animal{
	private static ArrayList<Anfibio>listado=new ArrayList<>();
	public static int ranas=0;
	public static int salamandras=0;
	private String colorPiel;
	private boolean venenoso;

	public Anfibio() {
    	super();
        Anfibio.listado.add(this);
	}
	
	public Anfibio(String nombre,int edad,String habitat,String genero,String colorPiel,boolean venenoso) {
		super(nombre,edad,habitat,genero);
		this.colorPiel = colorPiel;
		Anfibio.listado.add(this);
	
	}

	public  static ArrayList<Anfibio> getListado() {
		return Anfibio.listado;
	}
	public static void setListado(ArrayList<Anfibio> listado) {
		Anfibio.listado = listado;
	}
	public static int getRanas() {
		return ranas;
	}
	public static void setRanas(int ranas) {
		Anfibio.ranas = ranas;
	}
	public static int getSalamandras() {
		return salamandras;
	}
	public static void setSalamandras(int salamandras) {
		Anfibio.salamandras = salamandras;
	}
	public String getColorPiel() {
		return colorPiel;
	}
	public void setColorPiel(String colorPiel) {
		this.colorPiel = colorPiel;
	}
	public boolean isVenenoso() {
		return venenoso;
	}
	public void setVenenoso(boolean venenoso) {
		this.venenoso = venenoso;
	}
	public String movimiento() {
		return "saltar";	
}
	public static Anfibio crearRana(String nombre,int edad,String genero) {
		Anfibio rana=new Anfibio(nombre,edad,"selva",genero,"rojo",true);
		Anfibio.ranas++;
		return rana;
	}
	public static  Anfibio crearSalamandra(String nombre,int edad,String genero) {
		Anfibio salamandra=new Anfibio(nombre,edad,"selva",genero,"negro y amarillo",false);
		Anfibio.salamandras++;
		return salamandra;
	}	
	public static int cantidadAnfibios() {
		return Anfibio.listado.size();
	}
}

File: taller-5-python-ronalcastro97/zooAnimales/anfibio.py
from zooAnimales.animal import Animal

class Anfibio(Animal):
    _listado=[]
    ranas=0
    salamandras=0

    def __init__(self, nombre, edad, habitat, genero,colorPiel,venenoso):
        super().__init__(nombre, edad, habitat, genero)
        self._colorPiel=colorPiel
        self._venenoso=venenoso
        Anfibio._listado.append(self)
    
    def getColorPiel(self):
        return self._colorPiel
    
    def setColorPiel(self,colorPiel):
        self._colorPiel=colorPiel

    def isVenenoso(self):
        return self._venenoso
    
    def setVenenoso(self,venenoso):
        self._venenoso=venenoso

    @classmethod
    def crearRana(cls,nombre,edad,genero):
        cls.ranas+=1
        return cls(nombre,edad,"selva",genero,"rojo",True)
    
    @classmethod
    def crearSalamandra(cls,nombre,edad,genero):
        cls.salamandras+=1
        return cls(nombre,edad,"selva",genero,"negro y amarillo",False)
    
    @staticmethod
    def cantidadAnfibios():
        return len(Anfibio._listado)
    
    @staticmethod
    def movimiento():
        return "saltar"
    
